forge-ai/src/main/java/forge/ai/AiAttackController.java

```diff
@@ -1173,8 +1173,8 @@ public int compare(Pair<GameEntity, Integer> r1, Pair<GameEntity, Integer> r2) {
         while (!attritionalAttackers.isEmpty() && humanLife > 0 && attackRounds < 99) {
             // sum attacker damage
             int damageThisRound = 0;
-            for (int y = 0; y < attritionalAttackers.size(); y++) {
-                damageThisRound += attritionalAttackers.get(y).getNetCombatDamage();
+            for (Card attritionalAttacker : attritionalAttackers) {
+                damageThisRound += attritionalAttacker.getNetCombatDamage();
             }
```

Comment:
@tehdiplomat @Agetian is it worth it to change this into a sum function with Stream lib?

Reply:
does this work?
`damageThisRound = Aggregates.sum(attritionalAttackers, Card::getNetCombatDamage)`

Reply:
Probably, Aggregates is one of ours from before Java8 times (I think we could update that later?)

but I was more thinking about one of these:
https://www.baeldung.com/java-stream-sum

I think `Collectors` might be the most useful for us?

I tried this example
```java
/* Online Java Compiler and Editor */

import java.util.*;
import java.util.stream.Collectors;

public class HelloWorld{

static public class Damage
{
    protected int damage;
    public Damage(int i) {damage = i;}
    public int getDamage() { return damage;}
}
     public static void main(String []args){
         List<Damage> list = new ArrayList<>();
         list.add(new Damage(1));
         list.add(new Damage(2));
         list.add(new Damage(3));
        System.out.format("Damage: %d", list.stream().collect(
            Collectors.summingInt(Damage::getDamage)));
     }
}
```

Reply:
I was thinking the same thing too when I saw that one.

The IDE does have an inspection that can convert for loops into streams. It's off by default, though, and I can kinda see why - it's kinda subjective where you draw the line between the stream form being shorter and simpler vs dense and difficult.

This block comes out looking okay:
![image](https://github.com/user-attachments/assets/da4f526a-e8b8-489e-91cf-bef5098aa70b)

This one though...
![image](https://github.com/user-attachments/assets/b7d7da7c-99ec-4942-86b1-a698e1bd315f)
More concise, but arguably trickier to read, and very likely more awkward to edit if you wanted to do something more complicated within that for loop structure.

And this one goes from "hard to process" to "complete nightmare":
![image](https://github.com/user-attachments/assets/4d2ccabb-c7d7-4477-9862-bf04222cb5d4)

"for loop -> stream" was something I was considering for a second wave of this cleanup process, but evidently there'd need to be a rule of thumb for which ones to apply and which ones to leave alone. Number of terms in the stream? Length of the resulting line? Only do it for single statement loops? Or could just leave that one alone...

Reply:
yeah i think that is for later MR, right now, `Aggregates.sum` might be good enough

but i think from the examples of that java sum page, `Collectors.summingInt` is imo my favorite.

Reply:
Yea lets do that as a different PR. These changes look like a bunch of good cleanups for now.